src/components/BaseCard.vue

```diff
@@ -1,37 +1,37 @@
 <template>
   <div class="flex flex-col rounded-lg shadow-md overflow-hidden">
     <div class="flex-shrink-0">
-      <img class="h-48 w-full object-cover" :src="post.attributes.src || 'https://i.stack.imgur.com/y9DpT.jpg'">
+      <img class="h-48 w-full object-cover" :src="get(post, 'attributes.src', 'https://i.stack.imgur.com/y9DpT.jpg')">
     </div>
     <div class="flex-1 bg-white p-6 flex flex-col justify-between">
       <div class="flex-1">
         <p class="text-sm font-medium text-indigo-600">
           <span
-              v-for="(tag, index) in post.attributes.tags || []"
+              v-for="(tag, index) in get(post, 'attributes.tags', [])"
               :key="index"
               class="hover:underline mr-2">{{ tag.name }}</span>
         </p>
-        <router-link :to="`blogs/${post.attributes.slug}`" class="block mt-2">
+        <router-link :to="`blogs/${get(post, 'attributes.slug', 'not-found')}`" class="block mt-2">
           <p class="text-xl font-semibold text-gray-900">
-            {{ post.attributes.title }}
+            {{ get(post, 'attributes.title', '') }}
           </p>
           <p class="mt-3 text-base text-gray-500">
-            {{ post.attributes.content }}
+            {{ get(post, 'attributes.content', '') }}
           </p>
         </router-link>
       </div>
       <div class="mt-6 flex items-center">
         <div class="flex-shrink-0">
-            <img class="h-10 w-10 rounded-full" :src="post.relationships.creator.avatar || 'https://t4.ftcdn.net/jpg/03/46/93/61/360_F_346936114_RaxE6OQogebgAWTalE1myseY1Hbb5qPM.jpg'" alt="Avatar">
+            <img class="h-10 w-10 rounded-full" :src="get(post, 'relationships.creator.avatar', 'https://t4.ftcdn.net/jpg/03/46/93/61/360_F_346936114_RaxE6OQogebgAWTalE1myseY1Hbb5qPM.jpg')" alt="Avatar">
```

Comment:
Same here, you should use a computed method that return the value for src.
The get method that you use there is too long and not easy o read.